File: statemachine/src/main/java/org/nichel/sm/annotation/OnPerformAction.java
package org.nichel.sm.annotation;

import android.support.annotation.StringRes;

import java.lang.annotation.Retention;
import java.lang.annotation.Target;

import static java.lang.annotation.ElementType.METHOD;
import static java.lang.annotation.RetentionPolicy.RUNTIME;

@Target(METHOD)
@Retention(RUNTIME)
public @interface OnPerformAction {
    @StringRes String enter();
    @StringRes String exit();
}

File: statemachine/src/main/java/org/nichel/sm/annotation/OnPerformAction.kt
package org.nichel.sm.annotation

import android.support.annotation.StringRes

@Target(AnnotationTarget.FUNCTION, AnnotationTarget.PROPERTY_GETTER, AnnotationTarget.PROPERTY_SETTER)
@Retention(AnnotationRetention.RUNTIME)
annotation class OnPerformAction(@StringRes val enter: String, @StringRes val exit: String)
